### PR TITLE
[Dynamic Instrumentation] Fix line exploration tests

### PR DESCRIPTION
**Pull request by dudikeleti**
## Summary of changes
Line exploration tests should run for now only for protobuf

### CODE REVIEW

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (6089)](https://github.com/DataDog/dd-trace-dotnet/tree/8a771932a0bc492e2caeb32e0c0e195391e78054)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6089) - mean (71ms)  : 68, 74
     .   : milestone, 71,
    master - mean (73ms)  : 66, 79
     .   : milestone, 73,

    section CallTarget+Inlining+NGEN
    This PR (6089) - mean (1,107ms)  : 1085, 1129
     .   : milestone, 1107,
    master - mean (1,109ms)  : 1080, 1138
     .   : milestone, 1109,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6089) - mean (111ms)  : 105, 116
     .   : milestone, 111,
    master - mean (109ms)  : 106, 113
     .   : milestone, 109,

    section CallTarget+Inlining+NGEN
    This PR (6089) - mean (774ms)  : 752, 797
     .   : milestone, 774,
    master - mean (777ms)  : 753, 802
     .   : milestone, 777,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6089) - mean (94ms)  : 90, 98
     .   : milestone, 94,
    master - mean (94ms)  : 90, 99
     .   : milestone, 94,

    section CallTarget+Inlining+NGEN
    This PR (6089) - mean (726ms)  : 707, 746
     .   : milestone, 726,
    master - mean (734ms)  : 713, 755
     .   : milestone, 734,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6089) - mean (190ms)  : 187, 194
     .   : milestone, 190,
    master - mean (191ms)  : 188, 193
     .   : milestone, 191,

    section CallTarget+Inlining+NGEN
    This PR (6089) - mean (1,199ms)  : 1169, 1228
     .   : milestone, 1199,
    master - mean (1,195ms)  : 1171, 1219
     .   : milestone, 1195,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6089) - mean (276ms)  : 273, 280
     .   : milestone, 276,
    master - mean (277ms)  : 273, 281
     .   : milestone, 277,

    section CallTarget+Inlining+NGEN
    This PR (6089) - mean (942ms)  : 925, 959
     .   : milestone, 942,
    master - mean (936ms)  : 921, 951
     .   : milestone, 936,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6089) - mean (265ms)  : 261, 270
     .   : milestone, 265,
    master - mean (264ms)  : 261, 268
     .   : milestone, 264,

    section CallTarget+Inlining+NGEN
    This PR (6089) - mean (924ms)  : 905, 943
     .   : milestone, 924,
    master - mean (931ms)  : 902, 960
     .   : milestone, 931,

```

**Review by andrewlock — COMMENTED** [review]
I'm not quite sure why, but this seems to have broken all the exploration tests, seemingly in the variable generation step 🤔

**Comment by andrewlock** [comment]
## Benchmarks Report for tracer :snail:

Benchmarks for #6089 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663): 
* 1 benchmarks are **faster**, with geometric mean 1.232
* 1 benchmarks are **slower**, with geometric mean 1.210
* 1 benchmarks have **more** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.ActivityBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartStopWithChild`|net6.0|7.96μs|43.8ns|248ns|0.0156| 0.00779|0|5.43&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartStopWithChild`|netcoreapp3.1|9.71μs|50.8ns|313ns|0.0186| 0.00465|0|5.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartStopWithChild`|net472|16.1μs|51.2ns|198ns|1.03| 0.315|0.105|6.06&nbsp;KB|
|#6089|`StartStopWithChild`|net6.0|7.96μs|45.4ns|321ns|0.0167| 0.00836|0|5.42&nbsp;KB|
|#6089|`StartStopWithChild`|netcoreapp3.1|9.72μs|53.5ns|334ns|0.0184| 0.00918|0|5.62&nbsp;KB|
|#6089|`StartStopWithChild`|net472|16.4μs|73.5ns|285ns|1.01| 0.287|0.0819|6.07&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`WriteAndFlushEnrichedTraces`|net6.0|468μs|261ns|1.01μs|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|632μs|310ns|1.16μs|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`WriteAndFlushEnrichedTraces`|net472|832μs|359ns|1.39μs|0.417| 0|0|3.3&nbsp;KB|
|#6089|`WriteAndFlushEnrichedTraces`|net6.0|476μs|473ns|1.83μs|0| 0|0|2.7&nbsp;KB|
|#6089|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|640μs|286ns|1.07μs|0| 0|0|2.7&nbsp;KB|
|#6089|`WriteAndFlushEnrichedTraces`|net472|827μs|433ns|1.68μs|0.411| 0|0|3.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendRequest`|net6.0|195μs|1μs|4.6μs|0.188| 0|0|18.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendRequest`|netcoreapp3.1|217μs|1.24μs|9.35μs|0.206| 0|0|20.61&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendRequest`|net472|0.000514ns|0.000309ns|0.00111ns|0| 0|0|0&nbsp;b|
|#6089|`SendRequest`|net6.0|192μs|922ns|5.53μs|0.204| 0|0|18.45&nbsp;KB|
|#6089|`SendRequest`|netcoreapp3.1|220μs|1.25μs|8.68μs|0.226| 0|0|20.61&nbsp;KB|
|#6089|`SendRequest`|net472|0.000259ns|0.000234ns|0.000874ns|0| 0|0|0&nbsp;b|
</details>


<details><summary><strong>Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark</strong> - Same speed :heavy_check_mark: More allocations :warning:



#### More allocations :warning: in #6089

|  Benchmark         | Base Allocated | Diff Allocated | Change   | Change % |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark.WriteAndFlushEnrichedTraces&#8209;netcoreapp3.1 | 41.67 KB | 41.88 KB | 217 B | 0.52%

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`WriteAndFlushEnrichedTraces`|net6.0|569μs|2.68μs|12.3μs|0.551| 0|0|41.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|672μs|3.05μs|11.8μs|0.327| 0|0|41.67&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`WriteAndFlushEnrichedTraces`|net472|844μs|2.53μs|9.78μs|8.33| 2.5|0.417|53.3&nbsp;KB|
|#6089|`WriteAndFlushEnrichedTraces`|net6.0|546μs|1.41μs|5.09μs|0.566| 0|0|41.45&nbsp;KB|
|#6089|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|688μs|2.56μs|10.2μs|0.343| 0|0|41.88&nbsp;KB|
|#6089|`WriteAndFlushEnrichedTraces`|net472|877μs|3.76μs|14.1μs|8.3| 2.62|0.437|53.34&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`ExecuteNonQuery`|net6.0|1.31μs|1.61ns|6.22ns|0.0144| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`ExecuteNonQuery`|netcoreapp3.1|1.74μs|1.41ns|5.45ns|0.0138| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`ExecuteNonQuery`|net472|2.01μs|2.82ns|10.9ns|0.157| 0|0|987&nbsp;B|
|#6089|`ExecuteNonQuery`|net6.0|1.27μs|0.998ns|3.74ns|0.0139| 0|0|1.02&nbsp;KB|
|#6089|`ExecuteNonQuery`|netcoreapp3.1|1.82μs|0.922ns|3.57ns|0.0136| 0|0|1.02&nbsp;KB|
|#6089|`ExecuteNonQuery`|net472|2.02μs|2.83ns|10.9ns|0.157| 0|0|987&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`CallElasticsearch`|net6.0|1.24μs|0.259ns|0.969ns|0.0138| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`CallElasticsearch`|netcoreapp3.1|1.59μs|0.595ns|2.3ns|0.0135| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`CallElasticsearch`|net472|2.54μs|1.83ns|7.09ns|0.157| 0|0|995&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`CallElasticsearchAsync`|net6.0|1.23μs|0.705ns|2.73ns|0.0136| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`CallElasticsearchAsync`|netcoreapp3.1|1.63μs|0.899ns|3.36ns|0.0138| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`CallElasticsearchAsync`|net472|2.5μs|1.67ns|6.49ns|0.167| 0|0|1.05&nbsp;KB|
|#6089|`CallElasticsearch`|net6.0|1.29μs|1.59ns|6.16ns|0.0136| 0|0|976&nbsp;B|
|#6089|`CallElasticsearch`|netcoreapp3.1|1.48μs|0.461ns|1.79ns|0.0132| 0|0|976&nbsp;B|
|#6089|`CallElasticsearch`|net472|2.39μs|1.33ns|5.16ns|0.158| 0|0|995&nbsp;B|
|#6089|`CallElasticsearchAsync`|net6.0|1.32μs|0.975ns|3.78ns|0.0133| 0|0|952&nbsp;B|
|#6089|`CallElasticsearchAsync`|netcoreapp3.1|1.66μs|0.934ns|3.62ns|0.0141| 0|0|1.02&nbsp;KB|
|#6089|`CallElasticsearchAsync`|net472|2.61μs|1.91ns|7.41ns|0.167| 0|0|1.05&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`ExecuteAsync`|net6.0|1.25μs|0.791ns|2.96ns|0.0131| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`ExecuteAsync`|netcoreapp3.1|1.66μs|0.784ns|2.93ns|0.0126| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`ExecuteAsync`|net472|1.85μs|0.991ns|3.84ns|0.145| 0|0|915&nbsp;B|
|#6089|`ExecuteAsync`|net6.0|1.34μs|0.974ns|3.77ns|0.0134| 0|0|952&nbsp;B|
|#6089|`ExecuteAsync`|netcoreapp3.1|1.68μs|0.833ns|3.12ns|0.0126| 0|0|952&nbsp;B|
|#6089|`ExecuteAsync`|net472|1.74μs|0.523ns|2.02ns|0.145| 0|0|915&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendAsync`|net6.0|4.27μs|1.72ns|6.19ns|0.0298| 0|0|2.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendAsync`|netcoreapp3.1|5.02μs|2.03ns|7.58ns|0.0376| 0|0|2.76&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendAsync`|net472|7.76μs|4.41ns|17.1ns|0.497| 0|0|3.15&nbsp;KB|
|#6089|`SendAsync`|net6.0|4.09μs|2.11ns|8.15ns|0.0305| 0|0|2.22&nbsp;KB|
|#6089|`SendAsync`|netcoreapp3.1|5.12μs|1.18ns|4.42ns|0.0359| 0|0|2.76&nbsp;KB|
|#6089|`SendAsync`|net472|7.83μs|2ns|7.73ns|0.499| 0|0|3.15&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|net6.0|1.58μs|0.646ns|2.5ns|0.0229| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|netcoreapp3.1|2.12μs|0.738ns|2.66ns|0.0224| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|net472|2.53μs|0.74ns|2.87ns|0.249| 0|0|1.57&nbsp;KB|
|#6089|`EnrichedLog`|net6.0|1.5μs|1.38ns|5.33ns|0.0231| 0|0|1.64&nbsp;KB|
|#6089|`EnrichedLog`|netcoreapp3.1|2.3μs|2.33ns|9.04ns|0.0219| 0|0|1.64&nbsp;KB|
|#6089|`EnrichedLog`|net472|2.51μs|0.808ns|3.13ns|0.249| 0|0|1.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|net6.0|113μs|130ns|502ns|0.0566| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|netcoreapp3.1|118μs|189ns|732ns|0.0586| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|net472|146μs|190ns|735ns|0.655| 0.218|0|4.46&nbsp;KB|
|#6089|`EnrichedLog`|net6.0|112μs|148ns|573ns|0.0559| 0|0|4.28&nbsp;KB|
|#6089|`EnrichedLog`|netcoreapp3.1|118μs|178ns|667ns|0.0587| 0|0|4.28&nbsp;KB|
|#6089|`EnrichedLog`|net472|146μs|104ns|390ns|0.655| 0.218|0|4.46&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|net6.0|3.15μs|1.18ns|4.58ns|0.0315| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|netcoreapp3.1|4.22μs|1.84ns|7.12ns|0.0297| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|net472|4.79μs|0.811ns|3.03ns|0.319| 0|0|2.02&nbsp;KB|
|#6089|`EnrichedLog`|net6.0|3μs|0.774ns|3ns|0.0316| 0|0|2.2&nbsp;KB|
|#6089|`EnrichedLog`|netcoreapp3.1|4.22μs|1.17ns|4.4ns|0.0294| 0|0|2.2&nbsp;KB|
|#6089|`EnrichedLog`|net472|4.76μs|1.4ns|5.24ns|0.319| 0|0|2.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendReceive`|net6.0|1.28μs|0.547ns|2.05ns|0.0159| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendReceive`|netcoreapp3.1|1.76μs|2.08ns|8.04ns|0.0149| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`SendReceive`|net472|2.07μs|1.1ns|4.26ns|0.183| 0|0|1.16&nbsp;KB|
|#6089|`SendReceive`|net6.0|1.31μs|5.22ns|18.8ns|0.0162| 0|0|1.14&nbsp;KB|
|#6089|`SendReceive`|netcoreapp3.1|1.74μs|0.887ns|3.43ns|0.0155| 0|0|1.14&nbsp;KB|
|#6089|`SendReceive`|net472|2.25μs|1.37ns|5.31ns|0.183| 0.00113|0|1.16&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|net6.0|2.72μs|0.993ns|3.84ns|0.0217| 0|0|1.6&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|netcoreapp3.1|3.99μs|0.803ns|3.11ns|0.022| 0|0|1.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`EnrichedLog`|net472|4.35μs|2.4ns|9.31ns|0.323| 0|0|2.04&nbsp;KB|
|#6089|`EnrichedLog`|net6.0|2.85μs|1.33ns|5.13ns|0.0228| 0|0|1.6&nbsp;KB|
|#6089|`EnrichedLog`|netcoreapp3.1|3.91μs|1.07ns|4.13ns|0.0215| 0|0|1.65&nbsp;KB|
|#6089|`EnrichedLog`|net472|4.34μs|1.35ns|4.86ns|0.322| 0|0|2.04&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #6089

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SpanBenchmark.StartFinishSpan&#8209;net472 | 1.210 | 616.75 | 746.45 | 



#### Faster :tada: in #6089

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SpanBenchmark.StartFinishScope&#8209;net6.0 | 1.232 | 605.21 | 491.37 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartFinishSpan`|net6.0|398ns|0.438ns|1.7ns|0.00817| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartFinishSpan`|netcoreapp3.1|567ns|1.18ns|4.58ns|0.00776| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartFinishSpan`|net472|617ns|0.362ns|1.4ns|0.0917| 0|0|578&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartFinishScope`|net6.0|605ns|0.514ns|1.85ns|0.00967| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartFinishScope`|netcoreapp3.1|656ns|0.653ns|2.53ns|0.00952| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`StartFinishScope`|net472|796ns|0.99ns|3.83ns|0.104| 0|0|658&nbsp;B|
|#6089|`StartFinishSpan`|net6.0|400ns|0.233ns|0.872ns|0.00801| 0|0|576&nbsp;B|
|#6089|`StartFinishSpan`|netcoreapp3.1|559ns|0.43ns|1.61ns|0.00779| 0|0|576&nbsp;B|
|#6089|`StartFinishSpan`|net472|746ns|0.698ns|2.7ns|0.0917| 0|0|578&nbsp;B|
|#6089|`StartFinishScope`|net6.0|492ns|0.343ns|1.28ns|0.0098| 0|0|696&nbsp;B|
|#6089|`StartFinishScope`|netcoreapp3.1|668ns|0.44ns|1.71ns|0.00941| 0|0|696&nbsp;B|
|#6089|`StartFinishScope`|net472|850ns|0.64ns|2.48ns|0.104| 0|0|658&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`RunOnMethodBegin`|net6.0|609ns|0.389ns|1.51ns|0.00974| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`RunOnMethodBegin`|netcoreapp3.1|903ns|0.481ns|1.86ns|0.00951| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/efe3ae0b4f8af93b2769e6b5c9c04f9e76497663)|`RunOnMethodBegin`|net472|1.13μs|0.878ns|3.4ns|0.104| 0|0|658&nbsp;B|
|#6089|`RunOnMethodBegin`|net6.0|667ns|0.458ns|1.78ns|0.00972| 0|0|696&nbsp;B|
|#6089|`RunOnMethodBegin`|netcoreapp3.1|896ns|0.559ns|2.16ns|0.0098| 0|0|696&nbsp;B|
|#6089|`RunOnMethodBegin`|net472|1.1μs|0.512ns|1.92ns|0.104| 0|0|658&nbsp;B|
</details>

**Comment by dudikeleti** [comment]
> I'm not quite sure why, but this seems to have broken all the exploration tests, seemingly in the variable generation step 🤔

All investigation tests are now passed, I would appreciate your review.

File: .azure-pipelines/ultimate-pipeline.yml [hunk 1]
```diff
@@ -3168,11 +3168,11 @@ stages:
 
     - template: steps/restore-working-directory.yml
 
-    - script: tracer\build.cmd SetupExplorationTests -ExplorationTestUseCase $(explorationTestUseCase) -ExplorationTestName $(explorationTestName)
+    - script: tracer\build.cmd SetupExplorationTests --ExplorationTestUseCase $(explorationTestUseCase) --ExplorationTestName $(explorationTestName)
```
Comment (andrewlock) [inline .azure-pipelines/ultimate-pipeline.yml:3171]: doh! Nice catch 😄

**Review by andrewlock — APPROVED** [review]
Nice, thanks!

**Pull request merged**